go/vt/vttablet/tabletserver/query_engine.go

```diff
@@ -300,6 +295,13 @@ func (qe *QueryEngine) GetPlan(ctx context.Context, logStats *tabletenv.LogStats
 	qe.mu.RLock()
 	defer qe.mu.RUnlock()
 	statement, err := sqlparser.Parse(sql)
+	if !skipQueryPlanCache && !sqlparser.SkipQueryPlanCacheDirective(statement) {
+		if plan := qe.getQuery(sql); plan != nil {
+			logStats.CachedPlan = true
+			return plan, nil
+		}
+	}
+
 	if err != nil {
 		return nil, err
 	}
```

Comment:
I think we should not move getting from cached plan to here in vttablet
Reasons:
* vttablet receive a normalized query from vtgate so it can avoid query parsing here and look directly for cached plan and return - this avoids parsing overhead
* begin/commit.. other transactional things do not go through this code path so the chances of cache miss getting added are the least
* vttablet can use the skipQueryPlanCache directly at the top before parsing to check where it should look it in cached plan or not